GunicornWebWorker fails to reset SIGCHLD causing spurious "Worker exited" errors

### Describe the bug

The `aiohttp.worker.GunicornWebWorker` class overrides `init_signals()` but fails to reset SIGCHLD to SIG_DFL. This causes workers to inherit the gunicorn master arbiter's SIGCHLD handler, leading to spurious error logs when application code spawns subprocesses.

## Evidence

### 1. aiohttp/worker.py has incomplete fix
Lines 191-192 contain a comment about resetting signals, but NO implementation:

```python
# Reset signals so Gunicorn doesn't swallow subprocess return codes
# See: https://github.com/aio-libs/aiohttp/issues/6130
# ← CODE IS MISSING HERE!
```

### 2. Base Worker resets SIGCHLD correctly
`gunicorn/workers/base.py` properly resets SIGCHLD:

```python
SIGNALS = ["ABRT", "HUP", "QUIT", "INT", "TERM", "USR1", "USR2", "WINCH", "CHLD"]

def init_signals(self):
    # reset signaling
    for s in self.SIGNALS:
        signal.signal(s, signal.SIG_DFL)  # ← Resets SIGCHLD
```

### 3. aiohttp Worker doesn't call super()
`aiohttp/worker.py:160-193` overrides `init_signals()` but never calls `super().init_signals()`, so SIGCHLD is never reset.

### To Reproduce

1.Install gunicorn and aiohttp latest versions
2.Run this code from below, see usage inside
```
#!/usr/bin/env python3
"""
Reproduction of aiohttp GunicornWebWorker SIGCHLD inheritance bug.

Usage:
    python aiohttp_sigchld_bug.py          # Show the bug
    python aiohttp_sigchld_bug.py --fixed  # Show with fix

Bug: Workers inherit master's SIGCHLD handler, logging spurious errors
about subprocess PIDs as if they were worker PIDs.

Expected: Clean worker startup
Actual (without fix): ERROR logs like "Worker (pid:X) exited with code 1"
where X never appears in "Booting worker" logs.
"""
import sys

# Run with gunicorn when executed directly
if __name__ == "__main__":
    import os
    import subprocess
    import tempfile

    apply_fix = "--fixed" in sys.argv

    # Create a separate module that spawns subprocesses during import
    submodule_content = """
import subprocess
import time

# Spawn subprocesses during module import to trigger SIGCHLD
# This simulates real-world code that checks versions, calls git, dpkg, etc.
processes = []
for i in range(5):
    # Spawn subprocess that exits with code 1
    p = subprocess.Popen(['false'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    processes.append(p)

# Small delay creates window where:
# 1. Subprocesses exit and become zombies
# 2. SIGCHLD is delivered to worker
# 3. If worker has inherited arbiter's handler, it reaps OTHER workers' zombies
time.sleep(0.05)

# Clean up our zombies
for p in processes:
    p.wait()
"""

    # Create the main app module
    app_content = """
# Import submodule that spawns subprocesses (triggers bug)
import subprocess_module

from aiohttp import web

async def hello(request):
    return web.Response(text="Hello World\\n")

async def application():
    app = web.Application()
    app.router.add_get("/", hello)
    return app
"""

    # Create config with optional fix
    config_content = """
workers = 4
worker_class = "aiohttp.worker.GunicornWebWorker"
bind = "127.0.0.1:8765"
loglevel = "info"
"""

    if apply_fix:
        config_content += """
# Fix: Reset SIGCHLD in post_fork hook
import signal

def post_fork(server, worker):
    signal.signal(signal.SIGCHLD, signal.SIG_DFL)
"""

    # Create temp directory with all files
    tmpdir = tempfile.mkdtemp()

    with open(os.path.join(tmpdir, "subprocess_module.py"), "w") as f:
        f.write(submodule_content)

    with open(os.path.join(tmpdir, "app.py"), "w") as f:
        f.write(app_content)

    with open(os.path.join(tmpdir, "config.py"), "w") as f:
        f.write(config_content)

    try:
        print("=" * 70)
        if apply_fix:
            print("Running WITH fix (post_fork resets SIGCHLD to SIG_DFL)")
            print("Expected: Clean startup, no ERROR logs")
        else:
            print("Running WITHOUT fix (aiohttp bug)")
            print("Expected: ERROR logs about PIDs that never 'Booted'")
        print("=" * 70)
        print("\nLook for:")
        print("  [INFO] Booting worker with pid: XXXX")
        if not apply_fix:
            print("  [ERROR] Worker (pid:YYYY) exited with code 1")
            print("  where YYYY ≠ any XXXX (these are subprocess PIDs!)")
        print("\nPress Ctrl+C to stop\n")

        subprocess.run([
            sys.executable, "-m", "gunicorn",
            "-c", "config.py",
            "app:application"
        ], cwd=tmpdir)
    finally:
        import shutil
        shutil.rmtree(tmpdir)
```


### Expected behavior

No `[ERROR] Worker (pid:4155) exited with code 1.` in logs

### Logs/tracebacks

```python-traceback
[ERROR] Worker (pid:4155) exited with code 1
[ERROR] Worker (pid:4155) exited with code 1.
[ERROR] Worker (pid:4156) exited with code 1
[ERROR] Worker (pid:4156) exited with code 1.
[ERROR] Worker (pid:4157) exited with code 1
[ERROR] Worker (pid:4157) exited with code 1.
[ERROR] Worker (pid:4158) exited with code 1
[ERROR] Worker (pid:4158) exited with code 1.
```

### Python Version

```console
$ python --version
Python 3.12.12+
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.13.5
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author:
Author-email:
License: Apache-2.0 AND MIT
Location: /usr/local/lib/python3.12/dist-packages
Requires: aiohappyeyeballs, aiosignal, attrs, frozenlist, multidict, propcache, yarl
Required-by:
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 6.7.1
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache License 2.0
Location: /usr/local/lib/python3.12/dist-packages
Requires:
Required-by: aiohttp, yarl
```

### propcache Version

```console
$ python -m pip show propcache
Name: propcache
Version: 0.4.1
Summary: Accelerated property cache
Home-page: https://github.com/aio-libs/propcache
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /usr/local/lib/python3.12/dist-packages
Requires:
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.23.0
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /usr/local/lib/python3.12/dist-packages
Requires: idna, multidict, propcache
Required-by: aiohttp
```

### OS

ubuntu 22.04

### Related component

Server

### Additional context

pip show gunicorn
Name: gunicorn
Version: 25.3.0
Summary: WSGI HTTP Server for UNIX
Home-page: https://gunicorn.org
Author:
Author-email: Benoit Chesneau <benoitc@gunicorn.org>
License-Expression: MIT
Location: /usr/local/lib/python3.12/dist-packages
Requires: packaging
Required-by:

### Code of Conduct

- [x] I agree to follow the aio-libs Code of Conduct

--- a/aiohttp/worker.py
+++ b/aiohttp/worker.py
@@ -179,12 +179,11 @@ class GunicornWebWorker(base.Worker):  # type: ignore[misc,no-any-unimported]
         # by interrupting system calls
         signal.siginterrupt(signal.SIGTERM, False)
         signal.siginterrupt(signal.SIGUSR1, False)
+
         # Reset SIGCHLD to default so Gunicorn doesn't swallow subprocess
         # return codes. Without this, workers inherit the master arbiter's
         # SIGCHLD handler, causing spurious "Worker exited" errors when
         # application code spawns subprocesses.
-        # See: https://github.com/aio-libs/aiohttp/issues/6130
-        # See: https://github.com/aio-libs/aiohttp/issues/12327
         signal.signal(signal.SIGCHLD, signal.SIG_DFL)
 
     def handle_quit(self, sig: int, frame: FrameType | None) -> None:

--- a/aiohttp/worker.py
+++ b/aiohttp/worker.py
@@ -179,8 +179,13 @@ class GunicornWebWorker(base.Worker):  # type: ignore[misc,no-any-unimported]
         # by interrupting system calls
         signal.siginterrupt(signal.SIGTERM, False)
         signal.siginterrupt(signal.SIGUSR1, False)
-        # Reset signals so Gunicorn doesn't swallow subprocess return codes
+        # Reset SIGCHLD to default so Gunicorn doesn't swallow subprocess
+        # return codes. Without this, workers inherit the master arbiter's
+        # SIGCHLD handler, causing spurious "Worker exited" errors when
+        # application code spawns subprocesses.
         # See: https://github.com/aio-libs/aiohttp/issues/6130
+        # See: https://github.com/aio-libs/aiohttp/issues/12327
+        signal.signal(signal.SIGCHLD, signal.SIG_DFL)
 
     def handle_quit(self, sig: int, frame: FrameType | None) -> None:
         self.alive = False

